projects/cdk/services/directive-styles.service.ts

```diff
@@ -10,11 +18,26 @@ import {TUI_DEFAULT_RENDERER} from '@taiga-ui/cdk/tokens';
     providedIn: 'root',
 })
 export class TuiDirectiveStylesService {
+    private readonly map = new Map<Type<unknown>, unknown>();
```

Comment:
why not WeakMap?

Reply:
Would it make any difference here? 🤔

Reply:
I think this will protect against memory leaks at the time of HMR

Reply:
But the key is just a class, nobody stores it anywhere. Wouldn't that cause this `WeakMap` to drop the reference?

Reply:
just wondering what you think, okey

Reply:
I think we don't really need this and can even drop this map altogether 🙂 it would behave as if empty component was attached to the same selector and act naturally except the component is created manually and doesn't take up place of that single component that can match a selector. But I figured we might as well save some performance since styles are attached only the first time.